### PR TITLE
Add scraping of prometheuses

### PR DESCRIPTION
**Pull request by omegas27**
Towards https://github.com/giantswarm/giantswarm/issues/13223 and https://github.com/giantswarm/giantswarm/issues/13222

### CODE REVIEW

**Comment by omegas27** [comment]
Tested on Gauss
image

File: files/templates/additional-scrape-configs.template.yaml [hunk 1]
```diff
@@ -2228,4 +2228,33 @@
...
+  tls_config:
+    ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+    insecure_skip_verify: true
+  relabel_configs:
+  - source_labels: [__meta_kubernetes_pod_labelpresent_giantswarm_io_monitoring]
+    regex: (true)
+    action: keep
+  - source_labels: [__meta_kubernetes_service_labelpresent_operated_prometheus]
+    regex: (true)
+    action: keep
```
Comment (TheoBrigitte) [inline files/templates/additional-scrape-configs.template.yaml:2247]: I don't get this. We already have the managed app config which should catch this.

Comment (omegas27) [reply files/templates/additional-scrape-configs.template.yaml:2247]: I think the managed app configs is checking it the annotation giantswarm.io is present, not the label. Let me check

Comment (omegas27) [reply files/templates/additional-scrape-configs.template.yaml]: It seems we cannot change the service labels, only thse of the pods so we cannot manage it using the managed apps job at the moment

Comment (TheoBrigitte) [reply files/templates/additional-scrape-configs.template.yaml]: ok I see. annotation and label confused me, sorry.

**Review by TheoBrigitte — CHANGES_REQUESTED** [review]
Please merge `master` and update the unit test before merging this.

```
$ go test ./... -update
```

**Comment by omegas27** [comment]
I merged master

File: files/templates/additional-scrape-configs.template.yaml [hunk 2]
```diff
@@ -828,10 +828,18 @@
   relabel_configs:
   - source_labels: [__meta_kubernetes_service_annotationpresent_giantswarm_io_monitoring]
     regex: (true)
-    action: keep
-  - source_labels: [__meta_kubernetes_service_annotation_giantswarm_io_monitoring]
+    target_label: __tmp_keep_me
```
Comment (TheoBrigitte) [inline files/templates/additional-scrape-configs.template.yaml:831]: Why do we need this `__tmp_keep_me` label ?

File: files/templates/additional-scrape-configs.template.yaml [hunk 3]
```diff
@@ -382,7 +382,7 @@
     action: keep
 {{ else }}
   - source_labels: [__meta_kubernetes_pod_label_k8s_app]
-    regex: controller-manager
+    regex: scheduler
```
Comment (TheoBrigitte) [inline files/templates/additional-scrape-configs.template.yaml:385]: Nice catch :+1:

File: files/templates/additional-scrape-configs.template.yaml (hunk 2 above)
Comment (omegas27) [reply files/templates/additional-scrape-configs.template.yaml:831]: This allows us to scape both managed apps with the giantswarm.io/monitoring=true annotations and labels. This is basically a logical OR in relabelling syntax

**Review by TheoBrigitte — APPROVED** [review]
Nice job

**Pull request merged**